main/lsp/LSPTask.cc

```diff
@@ -249,6 +249,14 @@ LSPQueryResult LSPTask::queryByLoc(LSPTypecheckerDelegate &typechecker, string_v
     Timer timeit(config.logger, "setupLSPQueryByLoc");
     const core::GlobalState &gs = typechecker.state();
     auto fref = config.uri2FileRef(gs, uri);
+
+    if (!fref.exists() && config.isFileIgnored(config.remoteName2Local(uri))) {
+        auto error = make_unique<ResponseError>(
+            (int)LSPErrorCodes::RequestFailed,
```

Comment:
```suggestion
            (int)LSPErrorCodes::InvalidRequest,
```

i think it's better to use an existing error code that's already defined by the specification than to add a new one

Reply:
It is defined in the upcoming 3.17 version of the spec:

https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#errorCodes

Reply:
Yeah, my reasoning behind that code was the description of the error matches what was occurring here.

```javascript
        /**
	 * A request failed but it was syntactically correct, e.g the
	 * method name was known and the parameters were valid. The error
	 * message should contain human readable information about why
	 * the request failed.
	 *
	 * @since 3.17.0
	 */
	export const RequestFailed: [integer](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#integer) = -32803;
```

Reply:
ah gotcha, thanks!

Reply:
So it's not exactly specified, but my take on `RequestFailed` is that it means more like "we ran into an internal error, caught the exception, and are showing you this message"

In this case, the failure is in fact purely a function of the parameters.

I hold my opinion less strongly knowing that at least the newly added error code is in the spec, but I still think there's ground to use `InvalidParams` here

Reply:
Sure, that's fine with me. My primary concern was around the error message being displayed rather than the error code itself. I'll push up the change. 👍🏻